tests/std/tests/P0896R4_ranges_alg_search/test.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+//
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+using P = pair<int, int>;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::search(borrowed<false>{}, borrowed<true>{})), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::search(borrowed<true>{}, borrowed<false>{})), ranges::subrange<int*>>);
+
+struct instantiator {
+    static constexpr array<P, 8> pairs = {
+        P{0, 42}, P{1, 42}, P{2, 42}, P{3, 42}, P{4, 42}, P{5, 42}, P{6, 42}, P{7, 42}};
+    static constexpr array<int, 3> not_pairs = {2, 3, 4};
+
+    template <class Fwd1, class Fwd2>
+    static constexpr void call() {
+        const Fwd1 range1{pairs};
+        const Fwd2 range2{not_pairs};
+
+        // defaulted predicate and projections
+        {
+            auto result = ranges::search(range1, range1);
+            STATIC_ASSERT(same_as<decltype(result), ranges::subrange<ranges::iterator_t<Fwd1>>>);
+            assert(result.begin() == range1.begin());
+            assert(result.end() == range1.end());
+        }
+        {
+            auto result =
+                ranges::search(ranges::begin(range2), ranges::end(range2), ranges::begin(range2), ranges::end(range2));
+            STATIC_ASSERT(same_as<decltype(result), ranges::subrange<ranges::iterator_t<Fwd2>>>);
+            assert(result.begin() == range2.begin());
+            assert(result.end() == range2.end());
+        }
+
+        // explicit predicate

```

Comment:
Would their be value in testing an explicit predicate other than `ranges::equal_to` since that is also the default argument when no predicate is provided?  I'm not sure if there is precedent for this in the other ranges algorithms.

Reply:
Not much value: it would be hard to mess up the function signatures in such a way that the library used the default predicate when you passed a different predicate. That said, it's not too hard so why not.